cfgov/v1/blocks.py

```diff
@@ -194,3 +195,56 @@ class PlaceholderCharBlock(PlaceholderFieldBlock, blocks.CharBlock):
 class ReusableTextChooserBlock(SnippetChooserBlock):
     class Meta:
         template = '_includes/snippets/reusable_text.html'
+
+
+class FeaturedMenuContent(blocks.StructBlock):
+    draft = blocks.BooleanBlock(required=False, default=False)
+    link = atoms.Hyperlink(required=False)
+    body = blocks.RichTextBlock(required=False)
+    image = atoms.ImageBasic(required=False)
+
+
+class Link(blocks.StructBlock):
+    state = blocks.ChoiceBlock(choices=[
+        ('both', 'Live and draft'),
+        ('live', 'Live'),
+        ('draft', 'Draft')],
+        default='both',
+        help_text='Select state for this link. If draft, will only show '
+        'on sharing sites (like Content). If live, will only show '
+        'if not sharing site.')
+    link_text = blocks.CharBlock(required=True)
+    page_link = blocks.PageChooserBlock(
+        required=False,
+        help_text='Link to a page in Wagtail.')
+    external_link = blocks.CharBlock(
+        required=False,
+        max_length=1000,
+        default="#",
```

Comment:
What's the rationale for having this default value versus just an empty field?

Reply:
That's a good question. I think I added it because of the [logic in the mega menu template](https://github.com/cfpb/cfgov-refresh/blob/master/cfgov/jinja2/v1/_includes/organisms/mega-menu.html#L71) that shows links as disabled/greyed out if they don't have a url. In production we'd probably want to check that a link has either a page or external url before saving, though. I could add validation for that:

<img width="602" alt="screen shot 2018-01-11 at 3 27 38 pm" src="https://user-images.githubusercontent.com/778171/34852805-5c3f25d8-f6e5-11e7-97d2-29790fa1bc94.png">

Reply:
@virginiacc thanks; if that's easy to do it would be nice, but I don't want it to be a blocker for this PR. As a first-time user of the Wagtail interface to this module I was confused by the '#' and was worried it might show up as the actual link. Thinking about it a minute it makes sense that the page link would take precedence.

What about, as an easier change, modifying the help text for the external link field, to add something like "Only used if page link is left blank"?

Reply:
@chosak I'd started the link validation so I added it and updated the external url field helper text in https://github.com/cfpb/cfgov-refresh/pull/3396/commits/37b552f1f90590cf4e6d1b18b988a678dc8f4da1. Thanks for the suggestion!